src/components/ProjectToolbar/index.jsx

```diff
@@ -0,0 +1,93 @@
+import cls from 'classnames';
+import { useEffect, useState } from 'react';
+import ProjectStatusIcon from '../ProjectStatusIcon';
+import ProjectStatusDek from '../ProjectStatusDek';
+import {
+  flex, layout, margin, colors, typography as type,
+} from '../../theme';
+import ButtonsGroup from './ButtonsGroup';
+import store from '../../store';
+
+export default function ProjectToolbar({ selectedProject, isArchive }) {
+  const [projectDetail, setProjectDetail] = useState({});
+  const [status, setStatus] = useState(undefined);
+  const [timestamp, setTimestamp] = useState(undefined);
+
+  // Gets project details when selected project changes
+  useEffect(() => {
+    (async () => {
+      if (!isArchive) {
+        const project = await store.getProject(selectedProject);
+        setProjectDetail(project);
+      } else {
+        setProjectDetail(selectedProject);
+      }
+    })();
+  }, [selectedProject]);
+
+  // Listener for changes to a project in the store (like backup time)
+  useEffect(() => {
+    if (!isArchive) {
+      const {
+        stores: { PROJECTS },
+      } = store;
+      const unlisten = PROJECTS.onKeyChange(selectedProject, (e) => {
+        setProjectDetail(e);
+      });
+      return () => {
+        unlisten.then((f) => f());
+      };
+    }
+    return () => {};
+  }, [selectedProject]);
+
+  // Sets status and timestamp from project details
+  useEffect(() => {
+    if (projectDetail) {
+      const { isUploaded, isPublished, lastUploaded } = projectDetail;
+      if (isArchive) {
+        setStatus('archive');
+      } else if (isPublished) {
+        setStatus('published');
+        setTimestamp(lastUploaded);
+      } else if (isUploaded) {
+        setStatus('uploaded');
+        setTimestamp(lastUploaded);
+      }
+    }
```

Comment:
Another thing we shouldn't be doing twice and could handle in a selector if we had atoms.